### ISSUE TITLE
[Issue] Odysee "Original 1080p" videos won't fcast.

### ISSUE THREAD

**Issue by mnbvasd** [futo-org/grayjay-android#415]
When trying to fcast a non-HLS video from Odysee, the fcast service says it is unsupported.

Grayjay provides no feedback and will try to continue playing the previous video. It mixes up the displayed meta data with the previous video.

**Comment by mnbvasd** [comment]
Example video: https://odysee.com/cloud-connected-cameras-are-a-scam-vava#e4c214bd0ae2ce73443b52da6a2638d6b6b6acf6

When new videos are uploaded to Odysee, they're not transcoded to HLS straight away, so they often only have the original source.
These fail to cast.

For step 4, the problem with using that video to test or any non-new ones is that it will default to play the HLS video, then when you select the original it will fall back to the HLS version.

The linux fcast device says "No compatible source found for this media." At the top of the screen, and an "X" in the middle.

You can get the same behaviour by downloading the original video, cast something else, then try casting the download.

**Comment by mnbvasd** [comment]
![Screenshot_20231129-161228](https://github.com/futo-org/grayjay-android/assets/579059/4ff0111c-d238-4c21-b1d4-8525c8c15b53)
Here, I'm casting a video as normal, see the the video progress time.

![Screenshot_20231129-161246](https://github.com/futo-org/grayjay-android/assets/579059/bd624bef-5700-4db4-a7e3-01274253e00e)
Here I'm "watching" the downloaded video, you can see the video progress, which is nearly 20 minutes into an 8 minute video.The timestamps of both videos are 19:44/ This could be _any_ Odysee video which hasn't had HLS transcoded yet. It normally seems to take between 30 minute and 1 hour from upload.

![tv](https://github.com/futo-org/grayjay-android/assets/579059/a17cb71b-d944-4187-a6f2-8491238070fa)
Here is a photo of the TV, with the error on it.

**Comment by koen-futo** [comment]
Thank you for the details I will have another look.

**Comment by mnbvasd** [comment]
Thanks for putting so much effort into trying to reproduce this.

The source device is a FP4/CalyxOS
The target devices are x64 MacOS, and ARM64 Debian Bullseye.
The Fcast clients I've been using are the Electron ones, I've **not** tried an Android client. The behaviour is the same on both of them.

If the issue is a codec problem, it seems to effect both Mac and Linux clients.

I don't think the problem is with networking, I did a tcpdump of the traffic between the Mac and phone:
```
12:26:21.181786 IP 192.168.178.123.43002 > 192.168.178.143.46899: Flags [P.], seq 5:135, ack 1, win 128, options [nop,nop,TS val 2542746687 ecr 2097330955], length 130
E.....@.@..`...{.......3...&........r......
..<?}....{"container":"text/plain; charset=utf-8","url":"http://192.168.178.123:9999/video-5f37c64e-82a6-4c35-b22a-5a1a676699f1","time":0}
```
The .123 host is the phone, the .143 is the FCast receiver. I see this packet from the phone to the receiver, but I don't see any packets from the receiver to the phone on port 9999.

I've attached some logs from the FCast receiver on the Mac, with me trying to stream and download, which are both failing.


Another way I've found to reproduce the problem is by first playing the original clip on the phone, and then trying to FCast it.

[example-download.txt](https://github.com/futo-org/grayjay-android/files/13512019/example-download.txt)
[example-streaming.txt](https://github.com/futo-org/grayjay-android/files/13512020/example-streaming.txt)
